src/text-buffer.coffee

```diff
@@ -13,6 +13,40 @@ MarkerStore = require './marker-store'
 Patch = require './patch'
 {spliceArray, newlineRegex} = require './helpers'
 
+class SearchResultCallback
```

Comment:
Not sure about the name "callback" as it implies a function to me, whereas this is an object. What about just `SearchResult`? Or are you looking for some other word to indicate the control methods it provides?